storm-core/src/jvm/backtype/storm/scheduler/TopologyDetails.java

```diff
@@ -403,13 +415,23 @@ public void addResourcesForExec(ExecutorDetails exec, Map<String, Double> resour
      * Add default resource requirements for a executor
      */
     public void addDefaultResforExec(ExecutorDetails exec) {
+        Double topologyComponentCpuPcorePercent = Utils.getDouble(topologyConf.get(Config.TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT), null);
+        if (topologyComponentCpuPcorePercent == null) {
+            LOG.warn("default value for topology.component.cpu.pcore.percent needs to be set!");
```

Comment:
Actually, this would make a good unit test.  I think if we have deployed storm, we can expect all of the required defaults to be defined in defaults.yaml.

With the unit test, if I remove a required config entry from defaults.yaml, the unit test will fail explicitly.  This would not be something we need to handle every time we call this method at runtime.